service/storage_proxy.cc

```diff
@@ -5685,7 +5686,8 @@ inet_address_vector_replica_set storage_proxy::filter_for_query(
 }
 
 bool storage_proxy::is_alive(const gms::inet_address& ep) const {
-    return _gossiper.is_alive(ep);
+    auto* gossiper = &_remote->_gossiper;
+    return gossiper->is_alive(ep);
```

Comment:
Same here -- doesn't `is_alive` deserve being a method of `remote`, not `storage_proxy`?

Reply:
Check out https://github.com/scylladb/scylla/pull/11091
https://github.com/scylladb/scylla/pull/11091/commits/107e6daaf1c1d5f6a2afb00c4af351d37c0913c5